pkg/streamingpromql/engine_test.go

```diff
@@ -2087,61 +2170,69 @@ func TestAnnotations(t *testing.T) {
 			expectedInfoAnnotations:    []string{},
 		}
 	}
+	runAnnotationTests(t, testCases)
+}
 
-	opts := NewTestEngineOpts()
-	mimirEngine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), log.NewNopLogger())
-	require.NoError(t, err)
-	prometheusEngine := promql.NewEngine(opts.CommonOpts)
-
-	const prometheusEngineName = "Prometheus' engine"
-	engines := map[string]promql.QueryEngine{
-		"Mimir's engine": mimirEngine,
+func TestClassicHistogramAnnotations(t *testing.T) {
+	mixedClassicHistograms := `
+		series{host="a", le="0.1"}  2
+		series{host="a", le="1"}    1
+		series{host="a", le="10"}   5
+		series{host="a", le="100"}  4
+		series{host="a", le="1000"} 9
+		series{host="a", le="+Inf"} 8
+		series{host="a"}            {{schema:0 sum:5 count:4 buckets:[9 2 1]}}
+		series{host="b"}            1
+		series{host="c", le="abc"}  1
+		series{host="d", le="0.1"}  2
+		series{host="d", le="1"}    1
+		series{host="d", le="10"}   5
+		series{host="d", le="100"}  4
+		series{host="d", le="1000"} 9
+		series{host="d", le="+Inf"} 8
+	`
 
-		// Compare against Prometheus' engine to verify our test cases are valid.
-		prometheusEngineName: prometheusEngine,
+	testCases := map[string]annotationTestCase{
+		"bad bucket label warning": {
+			data:                       mixedClassicHistograms,
+			expr:                       `histogram_quantile(0.5, series{host="c"})`,
+			expectedWarningAnnotations: []string{"PromQL warning: bucket label \"le\" is missing or has a malformed value of \"abc\" for metric name \"series\" (1:25)"},
```

Comment:
[nit] Might be clearer to use raw strings here to avoid the need for all the escaping